zproject/backends.py

```diff
@@ -35,18 +37,21 @@ def password_auth_enabled(realm):
     return False
 
 def dev_auth_enabled():
+    # type: () -> bool
     for backend in django.contrib.auth.get_backends():
         if isinstance(backend, DevAuthBackend):
             return True
     return False
 
 def google_auth_enabled():
+    # type: () -> bool
     for backend in django.contrib.auth.get_backends():
         if isinstance(backend, GoogleMobileOauth2Backend):
             return True
     return False
 
 def common_get_active_user_by_email(email, return_data=None):
+    # type: (str, Dict[str, Any]) -> Union[UserProfile, None]
```

Comment:
See my note on using `Optional[UserProfile]` for this sort of thing -- you'll want to update that throughout this branch